### PR TITLE
WIP: Add conda support to ChemicalToolBoX

### PR DESCRIPTION
**Pull request by leobiscassi**
Hi @bgruening,
After we talk about _chemfp_ tools I saw what the following tools still doesn't support conda packages:
- [x] chemfp
- [x] confab
- [x] ctb_machine_learning
- [x] data_source
- [ ] fragmenter
- [ ] natural_product_likeness
- [x] openbabel
- [x] osra
- [ ] pgchem
- [x] silicos-it
- [x] simsearch
- [ ] spectral_hk

I start work to add support to conda packages in all _chemicaltoolbox_ tools, what do you think about this? Can I continue this work?

P.S.: Currently building conda recipe to chemfp 2012.12.1 used in mol2fps.xml

### CODE REVIEW

**Comment by bgruening** [comment]
That would be awesome. I started to work on it here: https://github.com/bgruening/galaxytools/pull/339
So a few tools are already converted, a few others I started to create PR ... please search in bioconda and here so see some PR that are not yet ready but maybe are a good start.

I also think that a few of these tools from you list should already work. How have you constructed the list?

Thanks!!!

**Comment by leobiscassi** [comment]
I made the list based on open pull requests and _tool_dependencies.xml_ in tools directories.
I look at PR [#339](https://github.com/bgruening/galaxytools/pull/339/) and some of the items in my list are wrong (e.g., openbabel tools, it's have the _tool_dependencies.xml_ but the file wasn't called in tool wrapper). :sweat_smile:
If dependency is available on bioconda channel and don't have a PR to update the tool, you suggest make the changes or not? :thinking: 

Thanks in advance

**Comment by bgruening** [comment]
My advice would be to look at this file: https://github.com/bgruening/galaxytools/blob/master/.tt_blacklist

All tools that are in needs to have conda dependencies. It can be that the dependency already exists in conda, in this case just remove the tool_dependency.xml and the entry from the blacklist.

**Comment by leobiscassi** [comment]
ok, thanks!

**Comment by leobiscassi** [comment]
@bgruening There is something wrong with travis tests? I got the following error:

`The command "conda create -y -q -c bioconda --name iuc_conda samtools=0.1.19 bcftools" failed and exited with 127 during`

**Comment by bgruening** [comment]
Yes, there is a problem with travis and/or planemo in all repositories :( I will try to figure out what is going on :(

**Comment by bgruening** [comment]
Travis is working again.

**Comment by bgruening** [comment]
To make planemo happy I restructured a little bit the directory hierarchy. Thanks for your work on this!

**Comment by bgruening** [comment]
Looks good so far. We need to update the test data. Do you want to do this? Do you know about the `panemo test --update_test_data` argument?

**Comment by leobiscassi** [comment]
Not yet, but I'll read the docs. Tomorrow I do this and submit a new commit.

File: chemicaltoolbox/chemfp/mol2fps.xml [hunk 1]
```diff
@@ -3,8 +3,6 @@
     <!--parallelism method="multi" split_inputs="infile" split_mode="to_size" split_size="10000" shared_inputs="" merge_outputs="outfile"></parallelism-->
     <requirements>
         <requirement type="package" version="1.1p1">chemfp</requirement>
-        <requirement type="package" version="2016.03.3">rdkit</requirement>
-        <requirement type="package" version="2.3.2">openbabel</requirement>
```
Comment (leobiscassi) [inline chemicaltoolbox/chemfp/mol2fps.xml]: Why these dependencies was removed? I don't follow you in this, please, can you explain?

File: chemicaltoolbox/silicos-it/align-it/align-it_create_db.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <tool id="ctb_alignit_create_db" name="Pharmacophore" version="0.1">
     <description>generation (Align-it)</description>
     <requirements>
-        <requirement type="package" version="1.0.1">align_it</requirement>
+        <requirement type="package" version="1.0.3">align_it</requirement>
```
Comment (bgruening) [inline chemicaltoolbox/silicos-it/align-it/align-it_create_db.xml]: Sure!

File: chemicaltoolbox/chemfp/mol2fps.xml (hunk 1 above)
Comment (bgruening) [inline chemicaltoolbox/chemfp/mol2fps.xml]: If you look at the chemfp package, openbabel and rdkit are already a dependency on chemfp, so you don't need to define them here. Even worse in this case chemfp was depending on a newer version of openbabel than we did specify here, ending up with two different openbabel versions in the same conda environment, which in the end confuses chemfp.

Comment (leobiscassi) [inline chemicaltoolbox/chemfp/mol2fps.xml]: oh yeah, thanks! :+1:

**Comment by leobiscassi** [comment]
Hi,
I took a look in `planemo test --update_test_data` documentation and I'm not sure if I understanded correctly, how's the correct way to use this command?

**Pull request closed without merging**